### ISSUE TITLE
fail connecting to server  on 127.0.0.1:19530

### ISSUE THREAD

**Issue by wangshuai-wuhan** [langgenius/dify#3107]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] Pleas do not modify this template :) and fill in all the required fields.

### Dify version

0.5.11

### Cloud or Self Hosted

Self Hosted (Docker)

### Steps to reproduce

2024-04-04 22:37:14 worker-1           | pymilvus.exceptions.MilvusException: <MilvusException: (code=2, message=Fail connecting to server on 127.0.0.1:19530. Timeout)>

2024-04-04 21:23:35 nginx-1            | 172.18.0.1 - - [04/Apr/2024:13:23:35 +0000] "GET /console/api/workspaces/current? HTTP/1.1" 502 559 "http://localhost/datasets/create" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36 Edg/123.0.0.0" "-"
2024-04-04 21:23:35 nginx-1            | 2024/04/04 13:23:35 [error] 31#31: *9 connect() failed (111: Connection refused) while connecting to upstream, client: 172.18.0.1, server: _, request: "GET /console/api/app-templates HTTP/1.1", upstream: "http://172.18.0.5:5001/console/api/app-templates", host: "localhost", referrer: "http://localhost/datasets/create"
2024-04-05 00:08:11 nginx-1            | 172.18.0.1 - - [04/Apr/2024:16:08:11 +0000] "GET /console/api/datasets/process-rule HTTP/1.1" 200 229 "http://127.0.0.1/datasets/create" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36 Edg/123.0.0.0" "-"
2024-04-05 00:08:11 nginx-1            | 172.18.0.1 - - [04/Apr/2024:16:08:11 +0000] "GET /console/api/workspaces/current/models/model-types/rerank HTTP/1.1" 200 2899 "http://127.0.0.1/datasets/create" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36 Edg/123.0.0.0" "-"
2024-04-05 00:08:11 worker-1           | [2024-04-04 16:08:11,931: ERROR/MainProcess] Failed to create new connection using: 147e82ddee034c08a9f58b77ee01a153
2024-04-05 00:08:11 nginx-1            | 172.18.0.1 - - [04/Apr/2024:16:08:11 +0000] "GET /console/api/workspaces/current/default-model?model_type=rerank HTTP/1.1" 200 15 "http://127.0.0.1/datasets/create" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36 Edg/123.0.0.0" "-"
2024-04-05 00:08:12 worker-1           | [2024-04-04 16:08:11,998: ERROR/MainProcess] Cleaned dataset when dataset deleted failed
2024-04-05 00:08:12 worker-1           | Traceback (most recent call last):
2024-04-05 00:08:12 worker-1           |   File "/app/api/tasks/clean_dataset_task.py", line 52, in clean_dataset_task
2024-04-05 00:08:12 worker-1           |     index_processor.clean(dataset, None)
2024-04-05 00:08:12 worker-1           |   File "/app/api/core/rag/index_processor/processor/paragraph_index_processor.py", line 67, in clean
2024-04-05 00:08:12 worker-1           |     vector = Vector(dataset)
2024-04-05 00:08:12 worker-1           |   File "/app/api/core/rag/datasource/vdb/vector_factory.py", line 23, in __init__
2024-04-05 00:08:12 worker-1           |     self._vector_processor = self._init_vector()
2024-04-05 00:08:12 worker-1           |   File "/app/api/core/rag/datasource/vdb/vector_factory.py", line 105, in _init_vector
2024-04-05 00:08:12 worker-1           |     return MilvusVector(
2024-04-05 00:08:12 worker-1           |   File "/app/api/core/rag/datasource/vdb/milvus/milvus_vector.py", line 51, in __init__
2024-04-05 00:08:12 worker-1           |     self._client = self._init_client(config)
2024-04-05 00:08:12 worker-1           |   File "/app/api/core/rag/datasource/vdb/milvus/milvus_vector.py", line 240, in _init_client
2024-04-05 00:08:12 worker-1           |     client = MilvusClient(uri=uri, user=config.user, password=config.password)
2024-04-05 00:08:12 worker-1           |   File "/usr/local/lib/python3.10/site-packages/pymilvus/milvus_client/milvus_client.py", line 64, in __init__
2024-04-05 00:08:12 worker-1           |     self._using = self._create_connection(uri, user, password, db_name, token, **kwargs)
2024-04-05 00:08:12 worker-1           |   File "/usr/local/lib/python3.10/site-packages/pymilvus/milvus_client/milvus_client.py", line 548, in _create_connection
2024-04-05 00:08:12 worker-1           |     raise ex from ex
2024-04-05 00:08:12 worker-1           |   File "/usr/local/lib/python3.10/site-packages/pymilvus/milvus_client/milvus_client.py", line 545, in _create_connection
2024-04-05 00:08:12 worker-1           |     connections.connect(using, user, password, db_name, token, uri=uri, **kwargs)
2024-04-05 00:08:12 worker-1           |   File "/usr/local/lib/python3.10/site-packages/pymilvus/orm/connections.py", line 355, in connect
2024-04-05 00:08:12 worker-1           |     connect_milvus(**kwargs, user=user, password=password, token=token, db_name=db_name)
2024-04-05 00:08:12 worker-1           |   File "/usr/local/lib/python3.10/site-packages/pymilvus/orm/connections.py", line 302, in connect_milvus
2024-04-05 00:08:12 worker-1           |     gh._wait_for_channel_ready(timeout=timeout)
2024-04-05 00:08:12 worker-1           |   File "/usr/local/lib/python3.10/site-packages/pymilvus/client/grpc_handler.py", line 134, in _wait_for_channel_ready
2024-04-05 00:08:12 worker-1           |     raise MilvusException(
2024-04-05 00:08:12 worker-1           | pymilvus.exceptions.MilvusException: <MilvusException: (code=2, message=Fail connecting to server on 127.0.0.1:19530. Timeout)>
2024-04-05 00:08:12 nginx-1            | 172.18.0.1 - - [04/Apr/2024:16:08:12 +0000] "POST /console/api/datasets/indexing-estimate HTTP/1.1" 200 1590 "http://127.0.0.1/datasets/create" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36 Edg/123.0.0.0" "-"
2024-04-05 00:08:12 nginx-1            | 172.18.0.1 - - [04/Apr/2024:16:08:12 +0000] "GET /console/api/version?current_version=0.5.11-fix1 HTTP/1.1" 200 120 "http://127.0.0.1/datasets/create" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36 Edg/123.0.0.0" "-"
2024-04-05 00:08:12 worker-1           | [2024-04-04 16:08:12,329: INFO/MainProcess] Task tasks.clean_dataset_task.clean_dataset_task[ac7baf8b-e51f-4dc4-b6b2-8c55c78a1c95] succeeded in 10.54672581900013s: None
2024-04-05 00:08:14 worker-1           | [2024-04-04 16:08:14,547: INFO/MainProcess] Task tasks.document_indexing_task.document_indexing_task[e2fac557-568c-4378-afcc-9c9c754ed203] received
2024-04-05 00:08:14 worker-1           | [2024-04-04 16:08:14,564: INFO/MainProcess] Start process document: 29fb0736-8a83-468d-884f-20775d061248

### ✔️ Expected Behavior

dify 能到连接到milvns 127.0.0.1/19530. 并且能读写数据

### ❌ Actual Behavior

连接不上，出现上述错误

**Comment by FarVision2** [comment]
Try this

_http://host.docker.internal:19530/_

You also have to have it turned on in Settings > General > Use the WSL 2 based engine > Add the *.docker.internal names to the host's etc/hosts file

this also works if you are using managers like Attu

**Comment by wangshuai-wuhan** [comment]
@FarVision2  真的起作用了，太感谢了，我为此困惑的都要放弃了，幸好得到的提醒，这次我终于成功了

**Comment by wangshuai-wuhan** [comment]
@FarVision2 It really worked, thank you so much. I was so confused and almost gave up, but luckily I got the reminder, and this time I finally succeeded.

**Comment by taowang1993** [comment]
The docker-compose.yaml file is missing ports for milvus.
https://github.com/milvus-io/milvus/blob/master/deployments/docker/standalone/docker-compose.yml
The official milvus docker-compose.yml file has ports for milvus-standalone.
But Dify docker-compose.yaml file does not have the ports. 
I did not change the MILVUS_HOST=127.0.0.1 variable. I justed add the missing ports.
And now it is working fine.
